hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -166,11 +166,15 @@ public class HoodieMetadataPayload implements HoodieRecordPayload<HoodieMetadata
   /**
    * HoodieMetadata record index payload field ids
    */
-  public static final String RECORD_INDEX_FIELD_PARTITION = "partition";
+  public static final String RECORD_INDEX_FIELD_PARTITION = "partitionName";
   public static final String RECORD_INDEX_FIELD_FILEID_HIGH_BITS = "fileIdHighBits";
   public static final String RECORD_INDEX_FIELD_FILEID_LOW_BITS = "fileIdLowBits";
   public static final String RECORD_INDEX_FIELD_FILE_INDEX = "fileIndex";
   public static final String RECORD_INDEX_FIELD_INSTANT_TIME = "instantTime";
+  public static final String RECORD_INDEX_FIELD_FILEID = "fileId";
+  public static final String RECORD_INDEX_FIELD_FILEID_ENCODING = "fileIdEncoding";
+  public static final int RECORD_INDEX_FIELD_FILEID_ENCODING_UUID = 0;
+  public static final int RECORD_INDEX_FIELD_FILEID_ENCODING_RAW_STRING = 1;
```

Comment:
Can someone switch from 0 to 1? Let's say the record key is still uuid but i want to store in string format after a few commits. Should we handle this in metadata payload merge?

Reply:
Maybe we don't need as we are always picking the latest payload. But, should we even allow such switching? Maybe add a validation?

Reply:
it is doable. every commit can switch. 
for eg, we can do one write w/ kafka connect. and then do another write using JavawriteClient directly. the internal config serves the purposes. it is not a table config for this reason

Reply:
got it